rplugin/python3/deoplete/sources/tag.py

```diff
@@ -50,5 +61,5 @@ def parse_tags(f):
 
     for l in f.readlines():
         candidates += p.findall(l)
-    return candidates
+    return list(set(candidates))
```

Comment:
I am wondering what’s the point of list(set())? If you need to randomize list of candidates there is `random.shuffle` which makes it *explicit* what you are doing and does not put unneded requirements on the elements of the list.